basic/src/main/java/com/c8y/ms/templates/basic/service/DeviceService.java

```diff
@@ -0,0 +1,132 @@
+     * fragmentType filter "c8y_IsDevice". In order to get the max page size use
+     * RestPageRequest.MAX_PAGE_SIZE on the list.
+     *
+     * @return List<String>, a list of existing devices and their names.
+     */
+    public List<String> getAllDeviceNames() {
+        List<String> allDeviceNames = new ArrayList<>();
+
+        try {
+            InventoryFilter inventoryFilter = new InventoryFilter();
+            inventoryFilter.byFragmentType(IsDevice.class);
+
+            ManagedObjectCollection managedObjectsByFilter = inventoryApi.getManagedObjectsByFilter(inventoryFilter);
+            List<ManagedObjectRepresentation> allObjects = Lists.newArrayList(managedObjectsByFilter.get(RestPageRequest.MAX_PAGE_SIZE).allPages());
+
+            for (ManagedObjectRepresentation managedObjectRepresentation : allObjects) {
+                allDeviceNames.add(managedObjectRepresentation.getName());
+            }
+        } catch (SDKException exception) {
+            LOG.error("Error while loading devices from Cumulocity", exception);
+        }
+
+        return allDeviceNames;
+    }
+
+    public Optional<ManagedObjectRepresentation> getDeviceRepresentation(final String deviceId) {
+        if (StringUtils.isEmpty(deviceId)) {
+            return Optional.absent();
+        }
+
+        try {
+            final ManagedObjectRepresentation deviceRepresentation = inventoryApi.get(new GId(deviceId));
+
+            return Optional.of(deviceRepresentation);
+        } catch (SDKException exception) {
+            LOG.error("Error occurred while loading device representation", exception);
+        }
+
+        return Optional.absent();
+    }
+
+    /**
+     * Create a new device in Cumulocity. This device will have the name and type coming from the initial request.
+     * In addition, by using the device-capability-model library the managedObject will be defined as a device
+     * (IsDevice.class) and agent (Agent.class).
+     *
+     * @param device provides the name and the type for the device, which should be created
+     * @return ManagedObjectRepresentation of the newly created device
+     */
+    public Optional<ManagedObjectRepresentation> createDevice(final Device device) {
+        final ManagedObjectRepresentation managedObjectRepresentation = new ManagedObjectRepresentation();
+        managedObjectRepresentation.setName(device.getName());
+        managedObjectRepresentation.setType(device.getType());
+        managedObjectRepresentation.set(new IsDevice());
+        managedObjectRepresentation.set(new Agent());
+
+        try {
+            final ManagedObjectRepresentation response = inventoryApi.create(managedObjectRepresentation);
+
+            return Optional.of(response);
+        } catch (SDKException exception) {
+            LOG.error("Error occurred while create a new device", exception);
+        }
+
+        return Optional.absent();
+    }
+
+    @Scheduled(fixedRate = 10000)
```

Comment:
I would add some further reading in code comment or README.md, spring supports more configurations for scheduled tasks (https://www.baeldung.com/spring-scheduled-tasks), parameterizing for example should we use on this example and have the configuration in application.properties

 ```suggestion
        @Scheduled(fixedDelayString = "${scheduled.delay.millis:60000}", initialDelayString = "${scheduled.delay.millis:1000}")
```

Reply:
Updated documentation